generic/tests/ping.py

```diff
@@ -58,76 +64,141 @@ def _get_loss_ratio(output):
             # Fallback to a hardcode host, eg:
             ext_host = default_host
 
-    counts = params.get("ping_counts", 100)
+    counts = params.get("ping_counts", 10)
     flood_minutes = float(params.get("flood_minutes", 10))
 
     packet_sizes = params.get("packet_size", "").split()
+    interval_times = params.get("interval_time", "").split()
 
-    for i, nic in enumerate(vm.virtnet):
-        ip = vm.get_address(i)
-        if ip.upper().startswith("FE80"):
-            interface = utils_net.get_neigh_attch_interface(ip)
-        else:
-            interface = None
-        nic_name = nic.get("nic_name")
-        if not ip:
-            logging.error("Could not get the ip of nic index %d: %s",
+    if ping_default:
+        for i, nic in enumerate(vm.virtnet):
+            ip = vm.get_address(i)
+            if ip.upper().startswith("FE80"):
+                interface = utils_net.get_neigh_attch_interface(ip)
+            else:
+                interface = None
+            nic_name = nic.get("nic_name")
+            if not ip:
+                test.fail("Could not get the ip of nic index %d: %s",
                           i, nic_name)
-            continue
-
-        error.base_context("Ping test on nic %s (index %d) from host"
-                           " side" % (nic_name, i), logging.info)
-        for size in packet_sizes:
-            error.context("Ping with packet size %s" % size, logging.info)
-            status, output = utils_test.ping(ip, 10, packetsize=size,
-                                             interface=interface, timeout=20)
+                continue
+
+            error_context.context("Ping test on nic %s (index %d) from host"
+                                  " side" % (nic_name, i))
+            for size in packet_sizes:
+                for interval_t in interval_times:
+                    logging.info("Ping with packet size %s" % size)
```

Comment:
Since there are different time intervals, you may  want to log the time interval here as well. 
eg.`logging.info("Ping with packet size %s" with %s interval  % (size, interval_t))`